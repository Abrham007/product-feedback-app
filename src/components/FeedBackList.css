.FeedBackList {
  width: 825px;
  min-height: 600px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  border-radius: 10px;
}

.FeedBackList--empty {
  align-items: center;
  justify-content: center;
  background: #fff;
  gap: 24px;
}

@media screen and (max-width: 1120px) {
  .FeedBackList {
    width: 689px;
    gap: 16px;
  }
}
