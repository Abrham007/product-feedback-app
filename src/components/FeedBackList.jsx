import FeedBackEmpty from "./FeedBackEmpty";
import "./FeedBackList.css";
import FeedBackPost from "./FeedBackPost";

function FeedBackList(props) {
  let isEmpty = props.listOfFeedback.length === 0;

  return (
    <div
      className={isEmpty ? "FeedBackList FeedBackList--empty" : "FeedBackList"}
    >
      {isEmpty ? (
        <FeedBackEmpty />
      ) : (
        props.listOfFeedback.map((feedback) => (
          <FeedBackPost key={feedback._id} {...feedback} />
        ))
      )}
    </div>
  );
}

export default FeedBackList;
