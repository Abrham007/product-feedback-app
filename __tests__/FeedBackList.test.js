import FeedBackList from "../src/components/FeedBackList";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("FeedBackList", () => {
  it("shows the correct JSX when empty", () => {
    const component = renderer.create(
      <BrowserRouter>
        <FeedBackList listOfFeedback={[]} />
      </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
