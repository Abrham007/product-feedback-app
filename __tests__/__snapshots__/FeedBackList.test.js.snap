// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedBackList shows the correct JSX when empty 1`] = `
<div
  className="FeedBackList FeedBackList--empty"
>
  <div
    className="FeedBackEmpty "
  >
    <img
      alt=""
      className="FeedBackEmpty__icon"
      height={136}
      width={130}
    />
    <div
      className="FeedBackEmpty__text"
    >
      <h2
        className="FeedBackEmpty__header"
      >
        There is no feedback yet.
      </h2>
      <p
        className="FeedBackEmpty__paragraph"
      >
        Got a suggestion? Found a bug that needs to be squashed? We love hearing about new ideas to improve our app.
      </p>
    </div>
    <a
      className="AddBtn"
      href="/add"
      onClick={[Function]}
    >
      <svg
        fill="#F2F4FE"
        height="12"
        viewBox="0 -960 960 960"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M417-417H166v-126h251v-251h126v251h251v126H543v251H417v-251Z"
        />
      </svg>
      <span>
        Add Feedback
      </span>
    </a>
  </div>
</div>
`;
